include/zio/zio_buf.h

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2019 Thomas Burdick <thomas.burdick@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file zio_buf.h
+ *
+ * @brief ZIO Buffer provides a pollable interface to a FIFO like buffer
+ *
+ * Device drivers implement this interface using either a software or
+ * hardware FIFO.
+ *
+ * When the zio_buf poll event is available is based on the watermark.
+ * When a zio_buf contains at least watermark number of datums.
```

Comment:
This seems a bit unclear, maybe it could be reworded? The last phrase seems kind of incomplete?

Reply:
Ok I've reworded this entire comment section to better, I think, reflect what I mean